diff --git a/node_modules/react-native-safe-area-context/android/build.gradle b/node_modules/react-native-safe-area-context/android/build.gradle
index 736822d..ff5b335 100644
--- a/node_modules/react-native-safe-area-context/android/build.gradle
+++ b/node_modules/react-native-safe-area-context/android/build.gradle
@@ -45,6 +45,12 @@ if (isNewArchitectureEnabled()) {
     apply plugin: "com.facebook.react"
 }
 
+
+def reactNativeArchitectures() {
+    def value = project.getProperties().get('reactNativeArchitectures')
+    return value ? value.split(',') : ['armeabi-v7a', 'x86', 'x86_64', 'arm64-v8a']
+}
+
 android {
     compileSdkVersion getExtOrDefault('compileSdkVersion', 30)
 
@@ -57,6 +63,8 @@ android {
         ndkVersion rootProject.ext.ndkVersion
     }
 
+
+
     defaultConfig {
         minSdkVersion getExtOrDefault('minSdkVersion', 16)
         targetSdkVersion getExtOrDefault('targetSdkVersion', 28)
@@ -82,6 +90,9 @@ android {
                 }
             }
         }
+         ndk {
+            abiFilters(*reactNativeArchitectures())
+        }
     }
 
     lintOptions{
